core/trino-spi/src/main/java/io/trino/spi/predicate/Range.java

```diff
@@ -215,24 +365,59 @@ private void checkTypeCompatibility(Marker marker)
         }
     }
 
-    @Override
-    public int hashCode()
+    int compareLowBound(Range other)
+    {
+        if (this.isLowUnbounded() || other.isLowUnbounded()) {
+            return Boolean.compare(!this.isLowUnbounded(), !other.isLowUnbounded());
```

Comment:
This is absolutly unreadable :)

Reply:
I copied this from `RangeView` where this is equally unreadable and I admit that. Can you pease help me improve?

Reply:
It was mostly complaining. 
IMO this would be more readable:
```
        if (this.isLowUnbounded() && !other.isLowUnbounded()) {
            return -1;
        }
        if (!this.isLowUnbounded() && other.isLowUnbounded()) {
            return 1;
        }
        if (this.isLowUnbounded() && other.isLowUnbounded()) {
            return 0;
        }
```

But given fact this is a private method, test covered, and not likely to change, you can as well leave it as is.